chips/sam4l/src/ast.rs

```diff
@@ -246,15 +263,17 @@ impl<'a> Alarm for Ast<'a> {
         unsafe { (*self.regs).cv.get() }
     }
 
-    fn set_alarm(&self, tics: u32) {
-        self.disable();
+    fn set_alarm(&self, mut tics: u32) {
         while self.busy() {}
         unsafe {
+            let now = (*self.regs).cv.get();
```

Comment:
`self.regs` is a `*const Registers`. You cannot refer to the field of `*const` directly, you have to either dereference the value first, or convert it to a reference (an `&` instead of a `*const`). Everywhere else, we use the pattern of converting to a reference at the top of the function, but the AST implementation is old enough that we don't do that.

We should fix it, but I wanted to make the change as minimal as possible for this PR, so just adhered to the style in this module.